File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maj = nums[0]
        count = 1
        for i in range(len(nums)):
            if maj == nums[i]:
                count += 1
            else:
                count -= 1
                if count == 0:
                    count = 1
                    maj = nums[i]
        return maj

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer,Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0;i < nums.length; i++){
            if(!map.containsKey(nums[i]))
                map.put(nums[i], 1);
            else{
                map.put(nums[i], map.get(nums[i]) + 1);
            }
        }
        for(Map.Entry<Integer,Integer> e : map.entrySet())
            if(e.getValue() > (nums.length / 2))
                return e.getKey();
        return -1;
    }
}